src/System.Linq/src/System/Linq/Repeat.cs

```diff
@@ -99,6 +99,18 @@ public List<TResult> ToList()
                 return list;
             }
 
+            public HashSet<TResult> ToHashSet(IEqualityComparer<TResult> comparer)
+            {
+                HashSet<TResult> hashSet = new HashSet<TResult>(_count, comparer);
+
+                for (int i = 0; i != _count; ++i)
+                {
+                    hashSet.Add(_current);
+                }
```

Comment:
There is the case where a comparer is poorly implemented (always returning differently). I made a test case for this: `WithRepeat_NeverEqualsComparer`. What you think?